### PR TITLE
Unbeatable ai gameplay

### PR DESCRIPTION
**Pull request by rianp**
## Changes made

1. add choose best move functionality to SuperComputerPlayer class
2. refactor create board to automatically make a 3x3 board when player chooses unbeatable ai as opponent 
3. add SuperComputerPlayer move message

Acceptance criteria

1. computer selects the best available space using the minimax algorithm
4. the board is correctly marked
5. only possible outcomes should be a draw or computer wins

## Comments
I had to decide between using the `get_board_winner_status` method in the `Board` class or the `evaluate` method in the `SuperComputerPlayer` class because both methods essentially perform the same task. However, I was not comfortable with how the `get_board_winner_status` method modified the original board array. I attempted to refactor the `get_board_winner_status` method in different ways, but I could not find a better solution than using the` evaluate` method that I had already created in the `SuperComputerPlayer` class.

[trello link](https://trello.com/c/tyvliys3)

### CODE REVIEW

File: game.py [hunk 1]
```diff
@@ -17,6 +17,9 @@ def play_round(self):
 
         if name == "Bot":
             move = current_player.make_move(self.board.get_board())
+        elif name == "Super Bot":
+            self.console.print_string("Watch in awe as Super Player Bot makes a really impressive move!")
+            move = current_player.make_move(self.board.get_board())
```
Comment (ranizilpelwar) [inline game.py:22]: This is hilarious :)

File: test_player.py [hunk 2]
```diff
@@ -91,3 +91,153 @@ def test_computer_player(self):
...
+                       'expected_move': 5},
+                      {'name': "ai should block opponent's vertical win",
+                       'board': [["o", "x", 3], ["o", 5, "x"], [7, 8, 9]],
+                       'expected_move': 7},
+                      {'name': "ai should block opponent's horizontal win",
+                       'board': [["x", "o", "x"], ["o", 5, "o"], [7, 8, "x"]],
+                       'expected_move': 5},
+                      {'name': "ai should choose the last move available if all but one move is left",
+                       'board': [["o", "x", "o"], ["x", "o", "x"], ["x", "o", 9]],
+                       'expected_move': 9}
```
Comment (ranizilpelwar) [inline test_player.py:122]: Now that you've got a solid understanding of test structure and naming, it really shows in these tests! 
I really like how you are explaining what behavior you are checking and helps with finding the edge cases.
Well done :)

**Review by ranizilpelwar — APPROVED** [review]

File: player.py [hunk 3]
```diff
@@ -37,3 +37,109 @@ def get_name(self):
...
+
+    def is_moves_left(self, board):
+        for row in range(len(board)):
+            for col in range(len(board)):
+                if isinstance(board[row][col], int):
+                    return True
+        return False
+
+    def evaluate(self, board):
+        computer_player = "x"
```
Comment (ranizilpelwar) [inline player.py:77]: Nice job separating the logic

File: test_player.py [hunk 4]
```diff
@@ -91,3 +91,153 @@ def test_computer_player(self):
             expected_output = "x"
             result = self.player.get_mark()
             self.assertEqual(result, expected_output)
+
+    def test_make_move(self):
+        test_cases = [{'name': "ai should take middle cell on first move",
```
Comment (ranizilpelwar) [inline test_player.py:96]: Nice job with behavior-driven tests

**Review by ranizilpelwar — APPROVED** [review]
